### DISCUSSION TITLE
API: Cable Terminations Type dcim.interface, dcim.powerport "That choice is not one of the available choices"

### DISCUSSION THREAD

**Discussion by metrax** [netbox-community/netbox#10388] [Q&A] [answered]
Hi,

i'm trying to get the termination data for a powerport via API, but i cannot filter by termination_type dcim.powerport.
I also tried an interface termination with dcim.interface -> same error.
![image](https://user-images.githubusercontent.com/1061866/190606967-4b9e7af2-d9a3-43b5-b440-0f181c0daa91.png)
 
What are the correct filter options?

I'm running netbox 3.3.3.

Thanks.

**Comment by candlerb** [comment, marked as answer]
Hmm, that's a good question. As you've observed, the api *response* includes those values:

```
...
            "id": 52,
            "url": "https://netbox.example.net/api/dcim/cable-terminations/52/",
            "display": "Cable #64 to ether9",
            "cable": 64,
            "cable_end": "B",
            "termination_type": "dcim.interface",   <<<
            "termination_id": 441,
...
```

The [filter source code](https://github.com/netbox-community/netbox/blob/v3.3.4/netbox/dcim/filtersets.py#L1635-L1639) doesn't help.  And that error message is not in the Netbox source, it comes from Django.

`termination_type` is defined [here](https://github.com/netbox-community/netbox/blob/v3.3.4/netbox/dcim/models/cables.py#L227-L232):

```
    termination_type = models.ForeignKey(
        to=ContentType,
        limit_choices_to=CABLE_TERMINATION_MODELS,
        on_delete=models.PROTECT,
        related_name='+'
    )
```

and `CABLE_TERMINATION_MODELS` [here](https://github.com/netbox-community/netbox/blob/v3.3.4/netbox/dcim/constants.py#L94-L108):

```
CABLE_TERMINATION_MODELS = Q(
    Q(app_label='circuits', model__in=(
        'circuittermination',
    )) |
    Q(app_label='dcim', model__in=(
        'consoleport',
        'consoleserverport',
        'frontport',
        'interface',
        'powerfeed',
        'poweroutlet',
        'powerport',
        'rearport',
    ))
)
```

Maybe it's the model ID then?

```
netbox=# select id from django_content_type where app_label='dcim' and model='interface';
 id
----
 17
(1 row)
```

Yep: adding `/?termination_type=17` to the query works for me.

That's not very user-friendly, especially since this number will vary from system to system - but you can get the value you need from `/api/extras/content-types/?app_label=dcim&model=interface`

**Reply by metrax** [reply]
Thanks for your help, i will use your workaround and hope that it will be fixed sometime ;-)
